### DISCUSSION TITLE
About redis licensing

### DISCUSSION THREAD

**Discussion by oringnam** [apache/airflow#38925] [General]
Redis has been licensed in such a way that only versions up to 7.x are available as open-source. Do you guys think this will affect the structure of the Airflow Celery setup?

As far as I know, the Celery executor uses Redis as the default storage commonly. Could Valkey, which was forked from redis, be a possible alternative?

## Ref
- https://redis.io/legal/licenses/ 
- https://www.linuxfoundation.org/press/linux-foundation-launches-open-source-valkey-community

**Comment by potiuk** [comment]
See here - i just started public discussion here: https://lists.apache.org/thread/4mjs2mqcgxswk96wwvw1rjc969lq5rl7 -> we had discussed it in PMC and my personal opinion is that for Airflow this is no change. Since redis is optional feature and there are alternatives, we do not have to change anything as far as project is concerned, 

But if you would like to contribute Valkey support, this is a great idea. And if there are other opinions - devlist discussion is the right place to express them.

Generally I don't think personally there will be any change until somone contributes Valkey - in which case we might consider changing the docs to recommend it by default.

Few clarifications:

* the change is from 7.4 onwards not 7.x -> 7.2 line (including security support for a long time apparently is still BSD licence)
* Airflow's Celery is not Redis by default - while we recommended it, rabbitmq is also tested alterntative - but likely yes if Valkey support is there, as a direct fork of Redis, it woudl likely be a better replacement.

As usual in open-source - contributions are most welcome. you could for example switch to Valkey, test it on your own and contribute it back once you confirm it working, using Redis support as a "blueprint".
